### PR TITLE
docs: update changelog for Permissionless ICS 

### PR DESCRIPTION
**Pull request by sainoe**
## Description

Closes: #XXXX

Add changelog entries for Permissionless ICS notably API changes.

TODO: fix links once #2171 is merged.

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct `docs:` prefix in the PR title
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/interchain-security/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] reviewed "Files changed" and left comments if necessary 
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] Confirmed the correct `docs:` prefix in the PR title
- [ ] Confirmed all author checklist items have been addressed 
- [ ] Confirmed that this PR only changes documentation
- [ ] Reviewed content for consistency
- [ ] Reviewed content for spelling and grammar
- [ ] Tested instructions (if applicable)
- [ ] Checked that the documentation website can be built and deployed successfully (run `make build-docs`)

### CODE REVIEW

File: .changelog/unreleased/api-breaking/provider/2171-permissionless-ICS.md.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+
+- Deprecate the `chain-id` parameter in favour of `consumer-id` for all transactions(TXs) and queries targeting a unique consumer chain. Below is a list highlighting the changes in the CLI commands. All commands assume the prefix `interchain-security-pd tx|q provider`.
+  - **TXs:**
+    | Command                                 | Message                             |
```
Comment (mpoke) [inline .changelog/unreleased/api-breaking/provider/2171-permissionless-ICS.md.md:4]: Do not use a table as it's formatted weird in Markdown. Just use bullet point lists, e.g., 

- `opt-in [consumer-id] [consumer-pubkey]` -- submit a [MsgOptIn](https://github.com/cosmos/interchain-security/blob/feat/permissionless/proto/interchain_security/ccv/provider/v1/tx.proto#L256)

File: .changelog/unreleased/api-breaking/provider/2171-permissionless-ICS.md.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    |-----------------------------------------|-------------------------------------|
+    | `assign-consensus-key [consumer-id] [consumer-pubkey]`  | [MsgAssignConsensusKey](https://github.com/cosmos/interchain-security/blob/feat/permissionless/proto/interchain_security/ccv/provider/v1/tx.proto#L256)             |
```
Comment (mpoke) [inline .changelog/unreleased/api-breaking/provider/2171-permissionless-ICS.md.md:6]: Wrong line. Also, use v5.2.0 as a stable version for the link.

File: .changelog/unreleased/api-breaking/provider/2171-permissionless-ICS.md.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    | `opt-in [consumer-id] [consumer-pubkey]`               | [MsgOptIn](https://github.com/cosmos/interchain-security/blob/feat/permissionless/proto/interchain_security/ccv/provider/v1/tx.proto#L256)                          |
+    | `opt-out [consumer-id]`                                | [MsgOptOut](https://github.com/cosmos/interchain-security/blob/feat/permissionless/proto/interchain_security/ccv/provider/v1/tx.proto#L256)                         |
+    | `set-consumer-commission-rate [consumer-id] [commission-rate]` | [MsgSetConsumerCommissionRate](https://github.com/cosmos/interchain-security/blob/feat/permissionless/proto/interchain_security/ccv/provider/v1/tx.proto#L295) |
+
+
+  - **Queries:**
+    | Command                                        | gRPC Endpoint                                              |
```
Comment (mpoke) [inline .changelog/unreleased/api-breaking/provider/2171-permissionless-ICS.md.md:13]: Same about the table. Replace with bullet point list. e.g., 

- Brief description of the query. Command: `consumer-genesis [consumer-id]`.  REST:  `/interchain_security/ccv/provider/consumer_genesis/{consumer_id}`.

File: .changelog/unreleased/api-breaking/provider/2171-permissionless-ICS.md.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    | `validator-provider-key [consumer-id] [consumer-validator-address]` | `/interchain_security/ccv/provider/validator_provider_addr/{consumer_id}/{consumer_address}`   |
+    | `consumer-opted-in-validators [consumer-id]`   | `/interchain_security/ccv/provider/opted_in_validators/{consumer_id}` |
+    | `consumer-validators [consumer-id]`            | `/interchain_security/ccv/provider/consumer_validators/{consumer_id}`      |
+    | `validator-consumer-commission-rate [consumer-id]` | `/interchain_security/ccv/provider/consumer_commission_rate/{consumer_id}/{provider_address}` |
+    | `all-pairs-valconsensus-address [consumer-id]` | `/interchain_security/ccv/provider/address_pairs/{consumer_id}` |
+
+
+- Deprecate the following queries and governance proposals:
+  - **Queries:**
+    | Command                                 | gRPC Endpoint                                                   |
```
Comment (mpoke) [inline .changelog/unreleased/api-breaking/provider/2171-permissionless-ICS.md.md:26]: ditto: replace with bullet point list.

File: .changelog/unreleased/api-breaking/provider/2171-permissionless-ICS.md.md [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    |-----------------------------------------|-----------------------------------------------------------------|    
+    | `list-start-proposals`                   | `/interchain_security/ccv/provider/consumer_chain_start_proposals` |
+    | `list-stop-proposals`                   | `/interchain_security/ccv/provider/consumer_chain_stop_proposals`  |
+    | `list-proposed-consumer-chains`         | `/interchain_security/ccv/provider/proposed_consumer_chains`       |
+  - **Proposals:**
+    - [ConsumerAdditionProposal](https://github.com/cosmos/interchain-security/blob/feat/permissionless/proto/interchain_security/ccv/provider/v1/provider.proto#L31)
```
Comment (mpoke) [inline .changelog/unreleased/api-breaking/provider/2171-permissionless-ICS.md.md:32]: ditto: use v5.2.0 version for links.

File: .changelog/unreleased/api-breaking/provider/2171-permissionless-ICS.md.md [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+
+- Deprecate the following queries and governance proposals:
+  - **Queries:**
+    | Command                                 | gRPC Endpoint                                                   |
+    |-----------------------------------------|-----------------------------------------------------------------|    
+    | `list-start-proposals`                   | `/interchain_security/ccv/provider/consumer_chain_start_proposals` |
+    | `list-stop-proposals`                   | `/interchain_security/ccv/provider/consumer_chain_stop_proposals`  |
+    | `list-proposed-consumer-chains`         | `/interchain_security/ccv/provider/proposed_consumer_chains`       |
+  - **Proposals:**
```
Comment (mpoke) [inline .changelog/unreleased/api-breaking/provider/2171-permissionless-ICS.md.md:31]: `ChangeRewardDenomsProposal` was also deprecated. Actually all legacy proposals were.

File: .changelog/unreleased/api-breaking/provider/2171-permissionless-ICS.md.md [hunk 7]
```diff
@@ -0,0 +1,36 @@
+
+- Deprecate the `chain-id` parameter in favour of `consumer-id` for all transactions(TXs) and queries targeting a unique consumer chain. Below is a list highlighting the changes in the CLI commands. All commands assume the prefix `interchain-security-pd tx|q provider`.
+  - **TXs:**
```
Comment (mpoke) [inline .changelog/unreleased/api-breaking/provider/2171-permissionless-ICS.md.md:3]:
```suggestion
- Add the  Permissionless ICS feature on the provider (as per 
  [ADR-019](https://cosmos.github.io/interchain-security/adrs/adr-019-permissionless-ics)),
  which entails the following api-breaking changes on the provider.
  ([\#2171](https://github.com/cosmos/interchain-security/pull/2171))
  
  - Deprecate the `chain-id` parameter in favour of `consumer-id` for all transactions and queries targeting a unique consumer chain. Below is a list highlighting the changes in the CLI commands. All commands assume the prefix `interchain-security-pd tx/q provider`.
    
    - **Transactions**
```

File: .changelog/unreleased/api-breaking/provider/2171-permissionless-ICS.md.md [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+    |-----------------------------------------|-----------------------------------------------------------------|    
+    | `list-start-proposals`                   | `/interchain_security/ccv/provider/consumer_chain_start_proposals` |
+    | `list-stop-proposals`                   | `/interchain_security/ccv/provider/consumer_chain_stop_proposals`  |
+    | `list-proposed-consumer-chains`         | `/interchain_security/ccv/provider/proposed_consumer_chains`       |
+  - **Proposals:**
+    - [ConsumerAdditionProposal](https://github.com/cosmos/interchain-security/blob/feat/permissionless/proto/interchain_security/ccv/provider/v1/provider.proto#L31)
+    - [ConsumerModificationProposal](https://github.com/cosmos/interchain-security/blob/feat/permissionless/proto/interchain_security/ccv/provider/v1/provider.proto#L140)
+    - [ConsumerRemovalProposal](https://github.com/cosmos/interchain-security/blob/feat/permissionless/proto/interchain_security/ccv/provider/v1/provider.proto#L122)  
+    
+([\#2171](https://github.com/cosmos/interchain-security/pull/2171))
```
Comment (mpoke) [inline .changelog/unreleased/api-breaking/provider/2171-permissionless-ICS.md.md:36]: Remove

File: .changelog/unreleased/features/provider/2171-permissionless-ICS.md.md
Comment (mpoke) [inline .changelog/unreleased/features/provider/2171-permissionless-ICS.md.md:1]: Similar feedback as for the other changelog entry.

**Review by mpoke — COMMENTED** [review]
Nice work @sainoe

File: .changelog/unreleased/api-breaking/provider/2171-permissionless-ICS.md.md (hunk 1 above)
Comment (sainoe) [reply .changelog/unreleased/api-breaking/provider/2171-permissionless-ICS.md.md]: Updated.

File: .changelog/unreleased/api-breaking/provider/2171-permissionless-ICS.md.md (hunk 2 above)
Comment (sainoe) [reply .changelog/unreleased/api-breaking/provider/2171-permissionless-ICS.md.md]: v5.2 doesn't have the new msgs definitions, i.e. MsgCreateConsumer. All links have to be fixed once we get a stable version with Permissionless.

File: .changelog/unreleased/api-breaking/provider/2171-permissionless-ICS.md.md (hunk 3 above)
Comment (sainoe) [reply .changelog/unreleased/api-breaking/provider/2171-permissionless-ICS.md.md]: Updated.

File: .changelog/unreleased/api-breaking/provider/2171-permissionless-ICS.md.md (hunk 2 above)
Comment (sainoe) [reply .changelog/unreleased/api-breaking/provider/2171-permissionless-ICS.md.md]: Changed to a main commit hash.

File: .changelog/unreleased/features/provider/2171-permissionless-ICS.md.md [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+      - `consumer-chain [consumer-id]`-- query details of a consumer chain associated with the consumer id.
+        - REST: `interchain-security/ccv/provider/consumer_chain/{consumer_id}`
+      - `consumer-id-from-client-id [client-id]` -- get the consumer id of a chain from a client id.
+        - REST: `interchain-security/ccv/provider/consumer_id/{client_id}`
+      - `blocks-until-next-epoch` -- query number of blocks remaining until the next epoch begins.
+        - REST: `interchain-security/ccv/provider/blocks_until_next_epoch`
+
+  - Improve the `list-consumer-chains` query to accept optional parameters `[phase]` and `[limit]`:
+    - `[phase]`: Filters returned consumer chains by their phase.
+    - `[limit]`: Limits the number of consumer chains returned.
```
Comment (mpoke) [inline .changelog/unreleased/features/provider/2171-permissionless-ICS.md.md:33]: @sainoe align with @MSalopek on this arg.

**Review by mpoke — APPROVED** [review]

File: .changelog/unreleased/api-breaking/provider/2171-permissionless-ICS.md.md [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+        - REST: `/interchain_security/ccv/provider/consumer_chain_start_proposals`
+
+      - `list-stop-proposals` -- consumer chains stop proposals on provider chain.
+        - REST: `/interchain_security/ccv/provider/consumer_chain_stop_proposals`
+
+      - `list-proposed-consumer-chains` -- query chain ids in consumer addition proposal before voting finishes.
+        - REST: `/interchain_security/ccv/provider/proposed_consumer_chains`
+
+    - **Proposals:**
+      - [ConsumerAdditionProposal](https://github.com/cosmos/interchain-security/blob/a17a3851b5eb3cec515b711dceae0afe9c14c3f0/proto/interchain_security/ccv/provider/v1/provider.proto#L31)
```
Comment (insumity) [inline .changelog/unreleased/api-breaking/provider/2171-permissionless-ICS.md.md:51]: Should we also mention that `MsgConsumerAddition`, `MsgConsumerRemoval`, etc. here as those are deprecated as well?

Comment (mpoke) [reply .changelog/unreleased/api-breaking/provider/2171-permissionless-ICS.md.md:51]: yeah, that's a good point. @sainoe could you please update.

**Pull request merged**